lib/nerves_hub_link/socket.ex

```diff
@@ -443,6 +443,19 @@ defmodule NervesHubLink.Socket do
     {:noreply, stop_iex(socket)}
   end
 
+  def handle_info({:update_manager, result}, socket) do
+    formatted =
+      if is_atom(result) do
+        %{result: result}
+      else
+        %{result: elem(result, 0), reason: elem(result, 1)}
```

Comment:
If we're not doing anything with the `{:error` portion of the tuple, why have it? We could just simplify and make it an atom return

Reply:
That or change the success condition to be `{:ok, result}` and we always return the same structure of `%{result: :ok | :error, reason: reason}`

Reply:
there are six events that can be sent from the `UpdateManager` to `Socket` are, the update ...

- started
- was ignored
- was rescheduled
- completed successfully
- failed due to a 401 error (`download_unauthorized`)
- failed due to a non fatal download error (`non_fatal`)

I had modelled this as ...
- `:started`
- `:ignored`
- `{:rescheduled, ms, rescheduled_to(ms)}`
- `:completed`
- `{:error, :download_unauthorized}`
- and `{:error, :non_fatal}`

I could change `:completed` to `{:ok, :completed}` to match the structure of the error cases, although the `:ok` is kinda superfluous information. It would also mean that `:ignored` and `:started` would need to change.

In Hub each of these cases would update registry in slightly different ways.

Thoughts?

I also realized that I hadn't accounted for `{:rescheduled, ms, rescheduled_to(ms)}`